modules/ROOT/pages/dataweave-extension-plugin.adoc

```diff
@@ -0,0 +1,362 @@
+. In Visual Studio Code, click *View > Command Palette*.
+. In the search bar, type `dataweave`.
+. Select *DataWeave: Create New Mapping*.
+
+image::dataweave-extension-createmapping.png[Command Palette with DataWeave: Create new mapping selected]
+
+
+[[sample-data]]
+=== Define Sample Data for DataWeave Mappings
+
+Generate multiple sample data inputs for your mapping by clicking the *Define Sample Data* button at the top of your mapping `.dwl` file. Sample data is a key part of DataWeave project development to run your mappings.
+
+To define sample data, follow these steps:
+
+. In Visual Studio Code, open your DataWeave project.
+. Navigate to the `src/main/dw` folder.
+. Select your mapping `.dwl` file.
+. Click the *Define Sample Data* button located above your mapping `.dwl` file.
++
+image::dataweave-extension-sampledata-1.png[MyMapping.dwl file with Define Sample Data button]
+
+
+Using sample data generates a DataWeave _scenario_ that contains all the resources required to run the mapping in the context of that scenario.
+
+image::dataweave-extension-sampledata-2.png[Generated new sample data payload.json file as input for the MyMapping.dwl file]
+
+[[scenarios]]
+== Understanding DataWeave Scenarios
+
+A DataWeave _scenario_ is a group of inputs and outputs that you can inject into your mapping either to try out in the *Preview Output* view or to evaluate your transformation and compare it with your desired output. Scenarios also provide autocompletion based on the input structures.
+
+These scenarios and their resources live in the `src/test/resources` folder and have the following structure:
+
+```
+├── pom.xml
+├── src
+│   ├── main
+│   │   ├── dw
+│   │   │   ├── MyMapping.dwl //Put your DataWeave file here
+│   │   └── resources
+│   └── test
+│       ├── dw
+│       │   └── Test.dwl //Unit tests go here
+│       │   └── MyMapping // Name of your mapping
+│       │       └── default //Name of your scenario
+│       │           └── inputs //Folder where all of your inputs live
+│       │           │   └── payload.json //Input for your mapping (you can use it in your script, naming it to be the same as the name of the file without the file's extension)
+│       │           │   └── vars //Folder that represents a name for your input variable in your script: in this case ,vars
+│       │           │       └── test.json //Input for your mapping (you can use it in your script, calling it vars.test)
+│       │           └── out.json // Expected output of your mapping running against your inputs (only present when testing)
+```
+=== Using the DataWeave Scenarios View
+
+The *DATAWEAVE SCENARIOS* view shows the available scenarios for the `.dwl` file. In this view, you can add, remove, or set the active scenario that is used for your preview execution, or you can choose that the values be autocompleted.
+
+image::dataweave-extension-dataweavescenario.png[DataWeave Scenarios view in Visual Studio Code]
+
+
+== Preview a DataWeave Mapping
```

Comment:
Remove this section if it is added previously in the suggestion of "Develop using the Preview"